services/appveyor.js

```diff
@@ -0,0 +1,39 @@
+const loadLogos = require('../lib/load-logos');
+
+module.exports = [{
+  name: 'AppVeyor',
+  type: 'build',
+  uri: {
+    format: '/appveyor/ci/([^/]+/[^/]+)(?:/(.+))?',
+    capture: ['repo', 'branch']
```

Comment:
I really like this idea. Group names definition is right below regex. Simple and useful implementation of named capturing groups.

Reply:
I feel like using `type` as property name might be misleading to developers. In the end, we all know that the `type` will define the left label. So, why no just call that property `leftLabel` or `title` ?

Reply:
The fact that it's used as the left label is an implementation detail. `type` is really the "group" that it falls under on the https://shields.io/ website. I could call it `group` or `category` instead maybe?

Reply:
I understand :smile: But I feel like forcing the semantic meaning of the left label as a "group" is too restrictive. What if users want to make tags that have a left label that is not an actual group ? As an example, yesterday I made (okay, it's not a great example, but it was quite fun) a badge that shows the weather in a given town (I haven't PR's it, I mean, not like anyone would really use it :stuck_out_tongue: ). I would not consider that "weather" is a type of badge. Anyways, if the goal is to keep it as a "type" semantic, I'd use it as `category` as I think developers could think that `type` is related to some sort of generic badge type ? (which I guess is you goal ^^).

Reply:
Agree `category` is good, since it suggests it's taxonomical in nature. Often the category is the same as the default label, so I like this way of handling it.

It's not a bad idea to provide a way to override the default label when it's different from the category. For example, the Github stars badge is in the "social" category but the default label is "Stars."